src/map.jl

```diff
@@ -119,3 +119,10 @@ end
 mapview(::typeof(identity), a) = a
 mapview(::typeof(identity), a::AbstractArray) = a
 mapview(::typeof(identity), d::AbstractDictionary) = d
+
+"""
+    filterview(f, a)
+
+Return a view of an array `a` without elements for which `f` is `false`. Similar to `filter(f, a)`, except returns a view instead of a copy.
+"""
+filterview(f, a::AbstractArray) = @view a[findall(f, a)]
```

Comment:
Oh yes, I see. Only partially lazy. I wasn't sure the best way to do this one.

Reply:
Yes, I don't think it's reasonably possible to make filtering completely lazy, unless we are talking about sequential iterators (`Iterators.filter`). For the result to be `getindex`able and `setindex`able, there is no way around precomputing filtered indices, as done here.

Reply:
Well it works for dictionaries. 

One idea I toyed with was making the `filterview` of arrays return a dictionary with a sparse set of integer keys. But it behaves quite differently to `filter` on arrays so what’s here is probably better.

Reply:
For dicts, (non)existence of a specific key doesn't affect the lookup for other keys as well. Arrays are different in this regard: if an element is filtered out, all further indices get shifted. So indeed, it's possible for dicts - but doesn't seem so for arrays.